pyanaconda/bootloader.py

```diff
@@ -2303,16 +2416,27 @@ def install(self, args=None):
 
 
 # every platform that wants a bootloader needs to be in this dict
-bootloader_by_platform = {platform.X86: GRUB2,
-                          platform.EFI: EFIGRUB,
-                          platform.MacEFI: MacEFIGRUB,
-                          platform.PPC: GRUB2,
-                          platform.IPSeriesPPC: IPSeriesGRUB2,
-                          platform.NewWorldPPC: MacYaboot,
-                          platform.S390: ZIPL,
-                          platform.Aarch64EFI: Aarch64EFIGRUB,
-                          platform.ARM: EXTLINUX,
-                          platform.omapARM: EXTLINUX}
+if flags.cmdline.get("legacygrub") == "1":
+    log.info("Using legacy grub (0.9x)")
+    bootloader_by_platform = {platform.X86: GRUB,
+                              platform.EFI: EFIGRUB1,
+                              platform.MacEFI: MacEFIGRUB,
+                              platform.PPC: GRUB2,
+                              platform.IPSeriesPPC: IPSeriesGRUB2,
+                              platform.NewWorldPPC: MacYaboot,
+                              platform.S390: ZIPL,
+                              platform.Aarch64EFI: Aarch64EFIGRUB}
+else:
+    bootloader_by_platform = {platform.X86: GRUB2,
+                              platform.EFI: EFIGRUB,
+                              platform.MacEFI: MacEFIGRUB,
+                              platform.PPC: GRUB2,
+                              platform.IPSeriesPPC: IPSeriesGRUB2,
+                              platform.NewWorldPPC: MacYaboot,
+                              platform.S390: ZIPL,
+                              platform.Aarch64EFI: Aarch64EFIGRUB,
+                              platform.ARM: EXTLINUX,
+                              platform.omapARM: EXTLINUX}
 
```

Comment:
I'm not opposed to that (it's how I would have written it) but this is just a revert of the diff that removed support for GRUB1 (with a few tweaks since the code is changed a bit). I was working to make it as close as possible to the previous code.... if the prefence is to change this in this PR, I can totally do that.